tests/v2/validation/provisioning/k3s/provisioning_node_driver.go

```diff
@@ -37,11 +37,16 @@ func TestProvisioningK3SCluster(t *testing.T, client *rancher.Client, provider P
 
 	machineConfigResp, err := client.Steve.SteveType(provider.MachineConfigPoolResourceSteveType).Create(machinePoolConfig)
 	require.NoError(t, err)
+	sshUser := machineConfigResp.JSONResp["sshUser"]
```

Comment:
I rather you create an extension to get the sshUser than have it here. What do you think @slickwarren ? Or to make this more expandable why not return the `machineConfigResp` instead, so we're not limiting it to just the sshUser?

Reply:
yes I agree.  though it would be cleaner to have an extension elsewhere, it does feel right that machineConfigResp should be returned here if we need it for some tests.

Reply:
I think once we get your PR in, that will become much clearer.

Reply:
I made the change to return `machineConfigResp` and also made few change to `pod_status.go` as it was throwing below error:
```
time="2023-08-08T12:10:39+05:30" level=info msg="Pod metrics-server-68cf49699b-ddcf4: Active | Image: docker.io/rancher/mirrored-metrics-server:v0.6.3"
E0808 12:10:39.599774   61814 runtime.go:79] Observed a panic: runtime.boundsError{x:0, y:0, signed:true, code:0x0} (runtime error: index out of range [0] with length 0)
goroutine 61 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x34529c0?, 0xc002b87b90})
	/home/infracloud/go/pkg/mod/k8s.io/apimachinery@v0.25.4/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0?})
	/home/infracloud/go/pkg/mod/k8s.io/apimachinery@v0.25.4/pkg/util/runtime/runtime.go:49 +0x75
panic({0x34529c0, 0xc002b87b90})
	/usr/local/go/src/runtime/panic.go:884 +0x212
github.com/rancher/rancher/tests/framework/extensions/workloads/pods.StatusPods.func1()
	/home/infracloud/go/src/github.com/rancher/rancher/tests/framework/extensions/workloads/pods/pod_status.go:49 +0x725
...
```